app/javascript/components/patient/close_contacts/CloseContactTable.js

```diff
@@ -0,0 +1,489 @@
+      }
+    );
+  };
+
+  /**
+   * Called when the number of entries to be shown on a page changes. Resets page to be 0.
+   * Updates state and then calls table update handler.
+   * @param {SyntheticEvent} event - Event when num entries changes
+   */
+  handleEntriesChange = event => {
+    const value = parseInt(event.target.value);
+    this.setState(
+      state => {
+        return {
+          query: { ...state.query, entries: value, page: 0 },
+        };
+      },
+      () => {
+        this.updateTable(this.state.query);
+      }
+    );
+  };
+
+  /**
+   * Called when the user clicks on the Contact Attempt for a Close Contact
+   */
+  handleContactAttempt = async data => {
+    if (await confirmDialog('Are you sure you want to log an additional contact attempt?', { title: 'New Contact Attempt' })) {
+      let currentCC = _.cloneDeep(this.state.table.rowData[`${data}`]);
+      currentCC['contact_attempts'] = currentCC['contact_attempts'] + 1;
+      this.setState(
+        {
+          showEditModal: false,
+          editRow: null,
+        },
+        () => {
+          this.updateCloseContact(currentCC, true);
+        }
+      );
+    }
+  };
+
+  /**
+   * Called when a page is clicked in the pagination component.
+   * Updates the table based on the selected page.
+   * @param {Object} page - Page object from react-paginate
+   */
+  handlePageUpdate = page => {
+    this.setState(
+      state => {
+        return {
+          query: { ...state.query, page: page.selected },
+        };
+      },
+      () => {
+        this.updateTable(this.state.query);
+      }
+    );
+  };
+
+  /**
+   * Toggle the Add New Close Contact modal by updating state.
+   */
+  toggleAddModal = () => {
+    this.setState({
+      showAddCloseContactModal: !this.state.showAddCloseContactModal,
+    });
+  };
+
+  /**
+   * Closes the Add New Close Contact modal and makes a request to add a new close contact to the db.
+   * @param {*} newCloseContactData - State from close contact modal containing needed close contact data.
+   */
+  handleAddSubmit = newCloseContactData => {
+    this.setState({ showAddCloseContactModal: false }, () => {
+      this.updateCloseContact(newCloseContactData, false);
+    });
+  };
+
+  /**
+   * Called when the edit close contact button is clicked.
+   * Updates the state to show the appropriate modal for editing a close contact.
+   */
+  toggleEditModal = row => {
+    this.setState({
+      showEditModal: true,
+      editRow: row,
+    });
+  };
+
+  /**
+   * Closes the Edit Close Contact modal and makes a request to update an existing close contact record.
+   * @param {*} updatedCloseContactData - State from close contact modal containing updated close contact data.
+   */
+  handleEditSubmit = updatedCloseContactData => {
+    const currentCloseContactId = this.state.table.rowData[this.state.editRow]?.id;
+    updatedCloseContactData['id'] = currentCloseContactId;
+    this.setState(
```

Comment:
you don't need the set state here because `updateCloseContact` reloads the page

Reply:
Disagree, im leaving this in for now.

Reply:
this one still needs to be updated

Reply:
done